hamilton/driver.py

```diff
@@ -1927,6 +1931,17 @@ def with_modules(self, *modules: ModuleType) -> "Builder":
         self.modules.extend(modules)
         return self
 
+    def with_functions(self, *functions: FunctionType) -> "Builder":
+        """Adds the specified functions to the list.
+        This can be called multiple times. If you have allow_module_overrides
+        set this will enabl overwriting modules or previously added functions.
+
+        :param functions:
+        :return: self
+        """
+        self.functions.extend(functions)
+        return self
```

Comment:
I don't think we want this? Why can't we just put it in the `.with_modules`?

E.g. this opens up issues of precedence when using both modules and functions.